salt/cloud/clouds/vmware.py

```diff
@@ -667,7 +671,8 @@ def _manage_devices(devices, vm=None, container_ref=None):
             adapter_type = devices['network'][network_adapter_label]['adapter_type'] if 'adapter_type' in devices['network'][network_adapter_label] else ''
             switch_type = devices['network'][network_adapter_label]['switch_type'] if 'switch_type' in devices['network'][network_adapter_label] else ''
             # create the network adapter
-            network_spec = _add_new_network_adapter_helper(network_adapter_label, network_name, adapter_type, switch_type, container_ref)
+            mac_address = devices['network'][device.deviceInfo.label].get('mac_address')
```

Comment:
Can this be changed to ``mac_address = devices['network'][network_adapter_label]['mac_address'] if 'mac_address' in devices['network'][network_adapter_label] else None``